lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,128 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+
+		if ($allUsers !== false) {
+			$users = [];
+			$function = function (IUser $user) use (&$users) {
+				$users[] = $user;
+			};
+
+			$this->userManager->callForAllUsers($function);
+			$this->setPasswordExpiryForUsers($input, $output, $users);
+		} elseif (\count($groups) >= 1) {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === false) {
+					$output->writeln("Group name $group doesn't exist");
+					return 1;
+				} else {
+					$users[$group] = [];
+					foreach ($this->getAllUsersFromGroup($group) as $users_array) {
+						$users[$group] = $users_array;
+						$this->setPasswordExpiryForUsers($input, $output, $users[$group]);
+					}
+				}
+			}
+		} else {
 
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
+			$uid = $input->getArgument('uid');
 
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
+			/** @var $user \OCP\IUser */
+			$user = $this->userManager->get($uid);
+
+			if ($user === null) {
+				$output->writeln("<error>Unknown user: $uid</error>");
+				return self::EX_NOUSER;
+			}
+
+			if (!$user->canChangePassword()) {
+				$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
+				return self::EX_GENERAL_ERROR;
+			}
+
+			if ($user !== null) {
+				$this->setPasswordExpiryForUsers($input, $output, [$user]);
+			}
 		}
 
-		if (!$user->canChangePassword()) {
-			$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
-			return self::EX_GENERAL_ERROR;
+		return 0;
+	}
+
+	protected function getAllUsersFromGroup($group) {
```

Comment:
this is overcomplicating things. The purpose of the generator is to avoid creating a large array in order to consume less memory. If the array is already created, wrapping it with a generator is pointless, and it's better (and easier) to pass the array along.

Reply:
Removed the usage of `getAllUsersFromGroup`. Iterating the output of `findUsersInGroup($group)`.